src/components/Buttons/BaseButton.tsx

```diff
@@ -47,6 +57,8 @@ function BaseButton({
   hasIcon = false,
   ...props
 }: BaseButtonProps) {
+  const [hitSlop, setHitSlop] = useState<null | Insets | number>({ top: 0, bottom: 0, left: 0, right: 0 });
```

Comment:
Could be better, create a `type`, and apply, it could be like:

```javascript
   type HitSlop = null | Insets | number
```

```suggestion
  const [hitSlop, setHitSlop] = useState<HitSlop>({ top: 0, bottom: 0, left: 0, right: 0 });
```